segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/HDFSExceptionHelpers.java

```diff
@@ -41,6 +42,8 @@
             return new StreamSegmentNotExistsException(segmentName, e);
         } else if (e instanceof FileAlreadyExistsException) {
             return new StreamSegmentExistsException(segmentName, e);
+        } else if (e instanceof EOFException) {
+            throw new ArrayIndexOutOfBoundsException(segmentName);
```

Comment:
This is an odd exception to throw in the case there is no more data to be read. I'd think that the goal of `convertException` is to convert low-level exceptions into a `StreamSegment*` exception that is meaningful to the segment store, otherwise I don't know why we would be doing this.

Reply:
This is following the contract at: https://github.com/pravega/pravega/blob/9bcfb22965933c50d5c22d55fb6ebd03482fc49c/segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/ReadOnlyStorage.java#L58. But it definitely looks sketchy. If you think this needs change, will create another issue for it.

Reply:
@fpj is right. The contract says it will throw that exception if the given buffer and bufferOffset & length are incorrect. It does not say anything about the file itself.

I disagree with blindly converting EOFExceptions into ArrayIndexOutOfBoundsExceptions. EOFExceptions can occur because we try to read beyond the end of an InputStream, which generally mean there's a bug somewhere else in our code. It will be almost impossible to pinpoint the source of the problem if all we see in the logs is an ArrayIndexOutOfBoundsException

Reply:
@arvindkandhare The contract you should be looking at is `SyncStorage`, not `ReadOnlyStorage`, as this is what this class implements.